pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -297,24 +297,15 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 							con.mu.Unlock()
 							continue
 						}
-					} else if discReq.ErrorDetail != nil {
```

Comment:
This is a case of version mismatch - I think we should still keep it. ErrorDetail can be present for variety of reasons. Version mismatch is one such case,